packages/node/src/integrations/localvariables.ts

```diff
@@ -19,6 +22,52 @@ type Next<T> = (result: T) => void;
 type Add<T> = (fn: Next<T>) => void;
 type CallbackWrapper<T> = { add: Add<T>; next: Next<T> };
 
+type RateLimitIncrement = () => void;
+
+/**
+ * Creates a rate limiter
+ * @param maxPerSecond Maximum number of calls per second
+ * @param enable Callback to enable capture
+ * @param disable Callback to disable capture
+ * @returns A function to call to increment the rate limiter count
+ */
+export function createRateLimiter(
+  maxPerSecond: number,
+  enable: () => void,
+  disable: (seconds: number) => void,
+): RateLimitIncrement {
+  let count = 0;
+  let retrySeconds = 5;
+  let disabledTimeout = 0;
+
+  setInterval(() => {
```

Comment:
why `setInterval` over `setTimeout`?

Reply:
This isn't really a "debounce", every second the exception count is checked and reset.

We could do a debounce with `setTimeout` but it feels like it would be more complex.